src/workflow/workflow.c

```diff
@@ -49,43 +38,24 @@ void workflow_confirm_dismiss(const char* title, const char* body)
         .title = title,
         .body = body,
     };
-    ui_screen_stack_switch(confirm_create(&params, _confirm_dismiss, NULL));
+    ui_screen_stack_switch(confirm_create(&params, _confirm_dismiss, NULL, NULL, NULL));
 }
 
-void workflow_start(void)
+workflow_t* workflow_allocate(workflow_method init, workflow_method cleanup, workflow_method spin)
 {
-#if PLATFORM_BITBOXBASE == 1
-    usart_start();
-    hww_setup();
-#elif PLATFORM_BITBOX02 == 1
-    usb_start(hww_setup);
-#endif
-    ui_screen_stack_pop_all();
-    ui_screen_stack_push(info_centered_create("See the BitBoxApp", NULL));
-}
-
-#if PLATFORM_BITBOX02 == 1
-/**
- * Called when the "select orientation" screen is over.
- * Switch to the main view.
- */
-static void _select_orientation_done(bool upside_down)
-{
-    if (upside_down) {
-        screen_rotate();
+    workflow_t* result = (workflow_t*)malloc(sizeof(*result));
```

Comment:
This seems to have surprised  everybody, but I've been doing this in PRs since forever :D This way if `result` changes type, there's no risk of crapping out the `malloc`'d size.